rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -228,6 +228,26 @@ bool ROSImageTexture::update()
   width_ = image->width;
   height_ = image->height;
 
+  // check if image is yuv format and convert it to rgb before normalizing data
+  if ( // NOLINT enforces bracket on the same line, which makes code unreadable
+    image->encoding == sensor_msgs::image_encodings::YUV422 ||
+    image->encoding == sensor_msgs::image_encodings::YUV422_YUY2)
+  {
+    // The number of bytes between the first pixel of the first row in memory
+    // and the first pixel of the second row in memory
+    uint32_t stride_in_bytes = image->step;
+    uint8_t * buffer = (uint8_t*)&image->data.front();
+
+    // yuv format uses 4 bytes to show 2 pixels in rgb format, which is 6 bytes
+    int new_size = (int)image->data.size() * 3 / 2;
+    uint8_t * dst_img  = new uint8_t[new_size];
```

Comment:
this will keep allocating memory everytime the function is called, which means that if you are getting frames at 33ms intervals a peice of the memory as big as one frame will get occupied and never freed. If this is done for enough time you will end up filling up your computers memory and the application will crash when new is unable to allocate anymore.

Do not allocate memory in runtime anyways because this will slow down the performance, use only static or preallocated memory. You can preallocate memory during initialization or as soon as you subscribe to the topic which I recommend.